wordpress-mobile/WordPress-iOS
# Feature/follow conversation tooltip

Refs pbArwn-4oo-p2

To test:
1. Enable local feature flag "Feature Highlight Tooltip".
2. Go to Reader, into any post.
3. A "New" anchor should appear (fade in) if the "Follow Conversation" button is not visible due to content size.
4. Tapping on it should scroll to the "Follow Conversation" button and the tooltip should animate in.
5. If "Follow Conversation" button is immediately visible, then anchor shouldn't appear as the button is already there.
6. The feature shouldn't be visible after the following: 
  - "Got it" button is tapped.
  - The tooltip is displayed 3 times without being interacted.
  - When the post is already subscribed/followed. 
  
The following snippet can be added to `ReaderDetailsViewController:viewDidLoad` to reset the local memory.
  ```Swift
          // FIXME: Remove, temporary code
        FeatureHighlightStore.didDismissTooltip = false
        FeatureHighlightStore.followConversationTooltipCounter = 0
  ```
  
Follow Up Points:
1. Sparkle emoji will be replaced with an asset for accessibility.
3. The tooltip should be dismissed when "Follow Conversation" button is tapped.
4. Add tracking
5. Refactor `FeatureHighlightStore` to make it testable and add unit tests.
6. Remove local feature flag

## Regression Notes
1. Potential unintended areas of impact
The additions aren't intrusive. There shouldn't be any regression.

8. What I did to test those areas of impact (or what existing automated tests I relied on)
`FeatureHighlightStore` will be refactored and unit tested in a follow up PR. The rest is UI code coupled to `ReaderDetailsViewController`. A big refactoring is needed to make things testable on that VC.

9. What automated tests I added (or what prevented me from doing so)
See 8.

PR submission checklist:
